### PR TITLE
Update version, fix warnings, create item with custom HTTP header fields

### PR DESCRIPTION
**Pull request by murselturk**

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -39,4 +39,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 39a3c51cfc7011b61bf96a8c38053b00ef42c157
 
-COCOAPODS: 1.5.2
+COCOAPODS: 1.5.3
```
Comment (raphrel) [inline Podfile.lock]: it seems you use your own pod binary.
pod is versionning in the Gemfile (currently 1.5.2), so can you update the cocopad gem in your PR ?
`bundle update cocoapods`

**Comment by raphrel** [comment]
I use to update changelog file at the same time, can you update it ? 
with ## [x.x.x] header

File: Podfile.lock (hunk 1 above)
Comment (murselturk) [reply Podfile.lock]: Done. https://github.com/noreasonprojects/ModernAVPlayer/pull/57/commits/e3e0de38464f42d4fa3ca11137913670f8de2551

**Comment by murselturk** [comment]
@raphrel I've updated the Changelog. Thank you for such a nice library :)

File: Sources/Core/State/LoadingMediaState.swift [hunk 2]
```diff
@@ -126,13 +126,19 @@ final class LoadingMediaState: PlayerState {
         context.player.replaceCurrentItem(with: nil)
     }
     
-    private func createItem(with url: URL) -> AVPlayerItem {
-        let asset = AVAsset(url: url)
+    private func createItem(with media: PlayerMedia) -> AVPlayerItem {
+        let asset: AVAsset
+        if let headerFields = media.headerFields {
+            let options = ["AVURLAssetHTTPHeaderFieldsKey": headerFields]
+            asset = AVURLAsset(url: media.url, options: options)
```
[the remark this answers is not in the document]
Comment (raphrel) [reply Sources/Core/State/LoadingMediaState.swift]: also, you can only use the avurlasset init:
`let asset = AVURLAsset(url: media.url, options: media.headerFields)`
another point to rename headerFields to assetOptions

**Pull request merged**